### DISCUSSION TITLE
[GitHub Enterprise Server] Cannot make a POST request to /repos/{owner}/{repo}/releases/{release_id}/assets to upload a release asset

### DISCUSSION THREAD

**Discussion by victoria-miltcheva** [community/community#64658] [Enterprise] [answered]
### Select Topic Area

Question

### Body

When using an [installation token](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation#using-an-installation-access-token-to-authenticate-as-an-app-installation) associated with a GitHub app, I can't make a request to the [upload release asset](https://docs.github.com/en/enterprise-server@3.3/rest/releases/assets#upload-a-release-asset) endpoint.

I'm on GitHub Enterprise Server 3.3.19.

I get the following when making a POST request to this endpoint and using the GitHub app installation token:

HTTP status: `502` Bad Gateway
Response body:
```
{"message":"Error uploading to https://media.github.host.com/releases/1376822/files: 404","request_id":"99dece88-4040-11ee-8a96-211186d12973"}
```
(I modified the host URL in the message value response)

However, I'm able to make the exact same request successfully when using my own GitHub PAT which has the `repo` scope:
![Screenshot 2023-08-22 at 8 28 46 AM](https://github.com/community/community/assets/12804086/0ba3efa0-79d0-4f11-a1fd-d5b207516d4c)

The GitHub app has Read & write permission for [Contents](https://docs.github.com/en/enterprise-server@3.3/rest/overview/permissions-required-for-github-apps#contents). Contents' description: `Repository contents, commits, branches, downloads, releases, and merges.`

However on the Permissions list for Contents, `POST /repos/{owner}/{repo}/releases/{release_id}/assets` isn't listed (endpoint for uploading a release asset). So this leads me to believe that a GitHub app installation token can't be used to call this endpoint. But on [the Endpoints available for GitHub Apps](https://docs.github.com/en/enterprise-server@3.3/rest/overview/endpoints-available-for-github-apps) page, `post /repos/{owner}/{repo}/releases/{release_id}/assets` is listed.

**My question is: is it possible to call the upload release asset endpoint with a GitHub app installation token?**

I found a similar question [here](https://stackoverflow.com/questions/68583389/github-app-permission-required-to-upload-an-asset-to-a-release).

**Comment by alriyadsouza** [comment]
It's possible that the endpoint for uploading release assets might not be fully supported by GitHub Apps, despite being listed in the documentation. Make sure your GitHub App has the necessary permissions for release-related actions. 
1)  Check GitHub App's Access: Confirm that the specific GitHub API endpoint you're trying to use (POST /repos/{owner}/{repo}/releases/{release_id}/assets) is supported for GitHub Apps. Refer to GitHub's official documentation and verify the compatibility.
2) Verify GitHub App Permissions: Double-check that your GitHub App has the appropriate permissions for uploading release assets. Ensure that it has explicit permissions related to releases or uploads, not just general "Contents" permissions.
If the issue continues consider checking with GitHub support for more assistance.

**Reply by victoria-miltcheva** [reply]
> It's possible that the endpoint for uploading release assets might not be fully supported by GitHub Apps, despite being listed in the documentation.
 
 This turned out to be correct (this is fixed in GHES 3.5), see https://github.com/orgs/community/discussions/64658#discussioncomment-7007796

**Comment by victoria-miltcheva** [comment, marked as answer]
Follow-up:

The issue was occurring because there is a bug with GitHub Enterprise Server where this endpoint doesn't work with GitHub app tokens. The bug is fixed in GHES 3.5.
